driver/docker/docker.go

```diff
@@ -417,27 +417,31 @@ func (d *Driver) validateImageDigest(image bundle.InvocationImage, repoDigests [
 		return nil
 	}
 
-	switch count := len(repoDigests); {
-	case count == 0:
+	if len(repoDigests) == 0 {
 		return fmt.Errorf("image %s has no repo digests", image.Image)
-	case count > 1:
-		return fmt.Errorf("image %s has more than one repo digest", image.Image)
 	}
 
-	// RepoDigests are of the form 'imageName@sha256:<sha256>'; we parse out the digest itself for comparison
-	repoDigest := repoDigests[0]
-	ref, err := reference.ParseNormalizedNamed(repoDigest)
-	if err != nil {
-		return fmt.Errorf("unable to parse repo digest %s", repoDigest)
-	}
-	digestRef, ok := ref.(reference.Digested)
-	if !ok {
-		return fmt.Errorf("unable to parse repo digest %s", repoDigest)
-	}
-	digest := digestRef.Digest().String()
+	for _, repoDigest := range repoDigests {
+		// RepoDigests are of the form 'imageName@sha256:<sha256>' or imageName:<tag>
+		// We only care about the ones in digest form
+		ref, err := reference.ParseNormalizedNamed(repoDigest)
+		if err != nil {
+			return fmt.Errorf("unable to parse repo digest %s", repoDigest)
```

Comment:
question: Should we error out here, or just ignore this case? 🤔 
I don't know if in this case we can have both digested reference AND tagged reference, but if we find the first one and just error out instead of trying the second one which is the one we want, we may return a validation error even if it's valid?

Reply:
Did you intend to comment on line 434 instead of where we check if the reference is in digest form and if not continue? This error check on line 429 is for when the repository digest is an invalid reference entirely. Or at least that is my intent.

Assuming you meant line 434, in practice there will always be a tagged and digested reference for the invocation images when Porter uses cnab-to-oci. When Porter first creates the invocation image, it pushes it to a registry, `myuser/mybundle-installer:v1.0`, this allows us to get a repository digest for the image. Then it is copied to its final location at `myuser/mybundle@sha256:invocation-image-repo-digest`. At runtime when we pull the invocation image and inspect it we get two repository digests back:

* `myuser/mybundle-installer:v1.0`
* `myuser/mybundle@sha256:invocation-image-repo-digest`

That's why I have the continue here on line 434. Apologies for the long tangent if that wasn't what you were asking! 😅

Reply:
@silvin-lubecki Let me know if that answers your question or if I need to make a change. Would love to merge this PR as it fixes a blocking bug in main. 😁

Reply:
Ok that makes sense, thanks @carolynvs 👍